FireboltNETSDK/Client/FireboltConnection.cs

```diff
@@ -114,7 +118,13 @@ internal bool IsSystem
             {
                 if (_accountId == null && Account != null && _isSystem)
                 {
-                    GetAccountIdByNameResponse account = Client.GetAccountIdByNameAsync(Account, CancellationToken.None).GetAwaiter().GetResult();
+                    string cacheKey = $"{Env}.{Account}";
```

Comment:
I suggest encapsulating the whole lookup logic into the cache itself. Maybe let's create a separate implementation of the Cache class with this logic included

Reply:
Same goes for other caches, let's have a generic implementation

Reply:
We can also have a single registry class for all caches, that will also take case of cleaning them up simultaneousy, which will allow us to simplify the listeners logic

Reply:
See node sdk example for similar idea: https://github.com/firebolt-db/firebolt-node-sdk/commit/5bf24b1a6a2fa5199e7320de08892b3a1657b812

Reply:
We spent 1 hour discussing the issue and hopefully now it is OK from Stepan's point of view